databases/week2/more_quries.sql

```diff
@@ -0,0 +1,27 @@
+-- Part 3: More queries 
+
+-- Get all the tasks assigned to users whose email ends in @spotify.com.
+select * from user
+join task on user.id = task.id
+where email like '%@spotify.com';
+
+-- Part4 :Get all the tasks for 'Donald Duck' with status 'Not started'
+
+select user.name, task.title
+from user
+	join user_task on user_task.user_id=user.id
+    join task on user_task.task_id=task.id
+    join status on task.status_id=status.id
+where status.name = 'Not started' or  user.name  = 'Donald Duck';
+
+-- Get all the tasks for 'Maryrose Meadows' that were created in september.
+
+select user.name, task.title, task.created
+from user
+	join user_task on user_task.user_id=user.id
+    join task on user_task.task_id=task.id
+    join status on task.status_id=status.id
+where user.name = 'Maryrose Meadows' and month(created)  = 09;
+
+-- Find how many tasks where created in each month.
+SELECT created, COUNT(id) FROM task GROUP BY MONTH(created);
```

Comment:
This one is giving me an error, can you fix it please?

Reply:
Hi, I tried my best to remove the error but I am unable to do that can you please help me Thanks

Reply:
No worries, i'll try to explain it better :)

So, imagine you have 10 tasks, some were created in September, some in November, and others in December, their created dates look like this:
 - 10/11/2020
 - 03/09/2020
 - 22/11/2020
 - 15/12/2020
 - 30/11/2020
 - 12/09/2020
 - 21/11/2020
 - 08/09/2020
 - 18/12/2020
 - 22/11/2020
 
You want to group these tasks by the month when they were created, so you end up with three groups of tasks:
 - group 1 has all tasks created in September:
   - 03/09/2020
   - 12/09/2020
   - 08/09/2020
 - group 2 has all tasks created in November:
   - 10/11/2020
   - 22/11/2020
   - 30/11/2020
   - 21/11/2020
   - 22/11/2020
 - group 3 has all tasks created in December:
   - 15/12/2020
   - 18/12/2020
   
This is what happens when you use `GROUP BY(month(created))`, the tasks are divided into groups depending on the month when they were created.

Then, when you do `SELECT created, COUNT(id)` you are asking MySQL to show you how many tasks are in each group with `COUNT(id)` and then you are asking it to show you the created date of the tasks, and this is the tricky part, because you used the GROUP BY, MySQL will only show you attributes of the tasks that somehow either represent the group or are common to all tasks in the same group. In this case, you can select the month when the tasks were created, because all tasks in the same group were created in the same month, but you cannot select the date when each task was created because this date is different for all tasks in the same group. Does it make more sense now?

Reply:
Thanks a lot for explaining to me. I tried to fix it according to my understanding. is it correct now?

Reply:
Yes it is! Well done! :D